lib/irb.rb

```diff
@@ -697,52 +697,49 @@ def encode_with_invalid_byte_sequence(str, enc)
     end
 
     def handle_exception(exc)
-      if exc.backtrace && exc.backtrace[0] =~ /\/irb(2)?(\/.*|-.*|\.rb)?:/ && exc.class.to_s !~ /^IRB/ &&
```

Comment:
I don't think we should change this. With the same example, it'll cause NoMethodError:

```
irb(main):001> raise StandardError.new('error').tap{def _1.backtrace()=nil}
#<StandardError: error>
backtraces are hidden because undefined method `[]' for nil:NilClass was raised when processing them
irb(main):002> 
```

Reply:
Yes, it will cause NoMethodError, and I think it's OK because it will be rescued and prints something.

This is the original issue https://bugs.ruby-lang.org/issues/9063 and pull-request https://github.com/ruby/ruby/pull/434.
Nil backtrace was a bug of ActionView::Template::Error, but backtrace check was introduced because
> irb shouldn't die even if caught Exception is something wrong (sorah wrote in bugs.ruby-lang.org)

`raise Exception.new.tap{def _1.message()=nil}` will raise NoMethodError.
`def _1.full_message(*)=nil` `def _1.full_message()='forgot to write arg'` also raise error.
I think nil backtrace is not a common case that IRB should specially handle.

Reply:
Fair enough 👍